### ISSUE TITLE
Lz4 Compressed binary is short than original

### ISSUE THREAD

**Issue by Nivek92** [OpenMined/PySyft#1704]
While working on another issue I noticed, that the compressed binary can be longer than the original which doesn't make sense.

As such I would suggest to only use the compressed version if it's shorter than the original. But then we would need to make sure decompression is only used when compression was really applied.

**Comment by amit-rastogi** [comment]
I am picking this up

**Comment by amit-rastogi** [comment]
@Nivek92 Yes I am looking into adding both the compression schemes as options into the library as per my discussion with @iamtrask yesterday.

**Comment by amit-rastogi** [comment]
@Nivek92 @iamtrask So as I understand there would be following tasks as part of this issue-
- add "zstd" as an option in the library
- add the functionality to decide whether to compress the input or not. Add corresponding logic in decompression as well.
-add unit tests

**Comment by Nivek92** [comment]
@amit-rastogi sounds good to me. I edited the original issue text to reflect these tasks
